### PR TITLE
FileFactory added

### PR DESCRIPTION
**Pull request by palash25**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/io/FileFactory.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    def get_line(self, line):
+        return linecache.getline(self.filename, line)
+
+    def file_as_list(self):
+        self.lines = linecache.getlines(self.filename)
+        self.linecount = len(self.lines)
+
+    def file_as_bytes(self):
+        with open(self.filename, 'rb') as fp:
+            return fp.read()
```
Comment (Makman2) [inline coalib/io/FileFactory.py]: File contents should be cached.

File: coalib/io/FileFactory.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    def file_as_string(self):
+        return self.file_as_bytes().decode()
```
Comment (Makman2) [inline coalib/io/FileFactory.py]: Should be cached as well ;)

File: coalib/io/FileFactory.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+@generate_eq('filename')
+class FileFactory:
+
+    def __init__(self, filename):
+        self.filename = filename
+        self.parent = os.path.abspath(os.path.join(self.filename, os.pardir))
+        self.stamp = os.path.getmtime(self.filename)
+
+    def get_line(self, line):
+        return linecache.getline(self.filename, line)
```
Comment (Makman2) [inline coalib/io/FileFactory.py]: `linecache` conflicts with your `open file` below. So in the worst case you load the file twice. Can you maybe write your own simple line cache?

File: coalib/io/FileFactory.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('filename')
+class FileFactory:
+
+    def __init__(self, filename):
+        self.filename = filename
+        self.parent = os.path.abspath(os.path.join(self.filename, os.pardir))
```
Comment (Makman2) [inline coalib/io/FileFactory.py]: Make this a property too like for `timestamp`? And `filename` as well?

File: coalib/io/FileFactory.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import linecache
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('filename')
```
Comment (Makman2) [inline coalib/io/FileFactory.py]: And `timestamp` as well ;)

File: coalib/io/FileFactory.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (palash25) [reply coalib/io/FileFactory.py]: > os.path.basename

I think you mean os.path.dirname

[the remark this answers is not in the document]
Comment (Makman2) [reply coalib/io/FileFactory.py]: Oh yeah correct :+1:

**Pull request closed without merging**